conda/core/repodata.py

```diff
@@ -413,7 +413,7 @@ def fetch_repodata(url, schannel, priority,
         mtime = getmtime(cache_path)
     except (IOError, OSError):
         log.debug("No local cache found for %s at %s", url, cache_path)
-        if use_cache:
+        if context.offline or use_cache:
```

Comment:
The test I wrote only fails if conda tries to fetch repodata even if context.offline is set, so it does not fail if we don't make the modification you suggest (or otherwise I guess the CI would have failed on this PR). I was thinking of adding another test which does the same thing for a `file://` URL and then makes sure that it *does* fetch repodata despite the `--offline` flag.

Alternatively we could consolidate both cases in the same test if we add a `file://`-based channel and then assert that it fetches repodata for the `file://` URL but not the `http(s)://` URL.